### PR TITLE
Features

### LINKED ISSUE #10823: Trigger events for access collections
Now that we have `ElggAccessCollection` extending `ElggData` we should trigger `create`, `update`,`delete` events and deprecate old hooks with weird naming convention

### PR DESCRIPTION
**Pull request by jeabakker**
- feat(core): added download response
   ref #11178
- chore(access_collections): removed unused leftover function
   ref #12663
- chore(access_collection): coding style
- feat(access_collection): trigger 'create', 'update' and 'delete' events
  Use the standard event names for access_collection related events
  fixes #10823
- chore(tests): correctly unregister systemlog events
- fix(core): prevent issues with unsaved data and system log

### CODE REVIEW

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreAccessCollectionsTest.php [hunk 1]
```diff
@@ -25,49 +23,6 @@ public function up() {
 		elgg()->session->setLoggedInUser($this->user);
 	}
 
-	public function testUpdateACL() {
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreAccessCollectionsTest.php:28]: don't we need these tests?

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreAccessCollectionsTest.php (hunk 1 above)
Comment (jeabakker) [reply engine/tests/phpunit/integration/Elgg/Integration/ElggCoreAccessCollectionsTest.php:28]: check the commit a8a9bf43b7d613d4bcd61a163ede66862d5efa4d

File: engine/classes/Elgg/Database/AccessCollections.php [hunk 2]
```diff
@@ -407,180 +407,101 @@ public function canEdit(int $collection_id, int $user_guid = null) {
...
-	 * @return int|false The collection ID if successful and false on failure.
+	 * @return bool
 	 */
-	public function create(string $name, int $owner_guid = 0, string $subtype = null): ?int {
-		$name = trim($name);
-		if (empty($name)) {
-			return null;
+	public function create(\ElggAccessCollection $acl): bool {
+		if (!empty($acl->id)) {
+			$this->update($acl);
```
[the remark this answers is not in the document]
Comment (jeabakker) [reply engine/classes/Elgg/Database/AccessCollections.php]: cross call is an easy fix for developers which call the internal function incorrectly. It's also tested now

File: engine/classes/Elgg/Database/AccessCollections.php [hunk 3]
```diff
@@ -407,180 +407,101 @@ public function canEdit(int $collection_id, int $user_guid = null) {
...
 	 * @return bool
 	 */
-	public function update($collection_id, array $new_members = []) {
-		$acl = $this->get($collection_id);
-
-		if (!$acl instanceof \ElggAccessCollection) {
-			return false;
+	public function update(\ElggAccessCollection $acl): bool {
+		if (empty($acl->id)) {
+			return $this->create($acl);
```
[the remark this answers is not in the document]
Comment (jeabakker) [reply engine/classes/Elgg/Database/AccessCollections.php:454]: see create comment

**Review by jdalsem — APPROVED** [review]

**Pull request merged**